src/server/message.rs

```diff
@@ -280,9 +305,10 @@ impl RawMessage {
             serde_json::from_str(msg).map_err(|_| jsonrpc::Error::parse_error())?;
 
         // Per JSON-RPC/LSP spec, Requests must have id, whereas Notifications can't
-        let id = ls_command
-            .get("id")
-            .map(|id| serde_json::from_value(id.to_owned()).unwrap());
+        let id = match ls_command.get("id") {
+                Some(id) => serde_json::from_value(id.to_owned()).unwrap(),
+                _ => Id::Null,
```

Comment:
A common idiom is to use [Option::unwrap_or_else](https://doc.rust-lang.org/std/option/enum.Option.html#method.unwrap_or_else) when supplying a default value when matching against `None`

Reply:
I read the documentation, and I am not sure if unwrap_or_else can be applied here. But after reading more documentation I found [Option::map_or](https://doc.rust-lang.org/std/option/enum.Option.html#method.map_or) and applied.

If unwrap_or_else can indeed be used here, could you explain how?

Anyway, thanks a lot for pointing it out, I learnt something new!